pjsip/src/pjsua-lib/pjsua_call.c

```diff
@@ -5576,26 +5576,54 @@ static void pjsua_call_on_tsx_state_changed(pjsip_inv_session *inv,
 		found_idx = pj_strtok(&input, &delim, &token, 0);
 		if (found_idx != input.slen) {
 		    /* Get signal/digit */
-		    const pj_str_t STR_SIGNAL = { "Signal=", 7 };
-		    const pj_str_t STR_DURATION = { "Duration=", 9 };
+		    const pj_str_t STR_SIGNAL = { "Signal", 6 };
+		    const pj_str_t STR_DURATION = { "Duration", 8 };
 		    char *val;
+		    pj_ssize_t idx;
+		    pj_ssize_t count_equal_sign;
 
 		    val = pj_strstr(&input, &STR_SIGNAL);
 		    if (val) {
-			info.digit = *(val+STR_SIGNAL.slen);
-			is_handled = PJ_TRUE;
+			idx = 0;
+			count_equal_sign = 0;
+			while ((*(val + STR_SIGNAL.slen + idx) == 32 || *(val + STR_SIGNAL.slen + idx) == 61) &&
+			    (val - input.ptr + STR_SIGNAL.slen + idx < input.slen)) {
```

Comment:
Looks much better now, just few minor suggestions.

Perhaps better to rearrange it like this:
```
while ((val - input.ptr + STR_SIGNAL.slen + idx < input.slen) && 
      (*(val + STR_SIGNAL.slen + idx) == 32 || *(val + STR_SIGNAL.slen + idx) == 61))
{
```
The updates are:
- coding style: align multiline condition, new line for bracket `{` when condition is long/multiline
- better check validity first before accessing/evaluating the pointer content.

Furthermore, as `val + STR_SIGNAL.slen + idx` seems to be repeatedly used, perhaps it'll be more readable to have a new temporary var, e.g:
```
char *p = val + STR_SIGNAL.slen;
while ((p - input.ptr < input.slen) &&  (*p == ' ' || *p == '=')) {
   ...
   ++p;
}
...
info.digit = *p;
```

Reply:
I've updated as your suggestion. Thanks!
https://github.com/Truong-Thanh-Quang/pjproject/commit/838091b37926032546f5f3fe32ff4fa2523b3b58